db/migrate/20221115101732_add_deactivations_table.rb

```diff
@@ -0,0 +1,10 @@
+class AddDeactivationsTable < ActiveRecord::Migration[7.0]
+  def change
+    create_table :location_deactivations do |t|
```

Comment:
Yes, I think that's the idea.
The way I'm thinking about it is that `reactivation_date` can only be added to an existing `location_deactivations` record that has `deactivation_date`. 

My reasoning behind it is that we deactivate a location first and then reactivate it. 
We can't deactivate a location twice before it gets reactivated either, so it should be easy to track which record we add the reactivation to (which would be the only record with a missing reactivation date).

Let me know if that makes sense to you and if you've got different ideas on how we can approach this!

Reply:
I guess this could work, the only thing I'm worried about atm is whether we might need to associate which reactivation belongs to which deactivation for any reason.

I was thinking about a single table with a type column for location/scheme but ended up going down the separate tables route cause it was somewhat more simple to understand.

While I can see the benefit of generalising things, could it be a bit premature to do it at the moment? Are there any other events we expect to save?
I'd like to hear how it's more optimal to save it that way, cause  I'm struggling to see it atm

Reply:
So, I thought that both values would be set and that would constitute a deactivation period.
So a record like this:

```
id | deactivation_date | reactivation_date | location_id
1 |  2022-8-9 | 2022-8-20 | 3
```

would meant that the location with id 3 was not active between the 9th of August and 20th of August 2022.

If there's a record like this:

```
id | deactivation_date | reactivation_date | location_id
1 |  2022-8-9 | 2022-8-20 | 3
2 |  2022-8-23 | nil | 3
```

would meant that the location with id 3 was not active between the 9th of August and 20th of August 2022. And then got deactivated again on the 23rd of August and is currently deactivated

Reply:
I can see what you mean with the table name. I feel like it's clear to me cause I'm viewing deactivations as periods, but maybe that's not how everyone sees them. Is `location_deactivation_periods` clearer? Or would naming it something else make sense?

Reply:
Just to chip in I like `location_deactivation_periods` but keeping `deactivation_date` and `reactivation_date` as that's more similar to the mental model implied by the UI (as you separately add deactivation and reactivation dates rather than submitting a whole period at a time)

Reply:
Renamed it to that for now, thanks both